devel/git/Portfile

```diff
@@ -73,13 +69,29 @@ build.args          CFLAGS="${CFLAGS}" \
                     CURLDIR=${prefix} \
                     OPENSSLDIR=${prefix} \
                     ICONVDIR=${prefix} \
-                    PERL_PATH="${prefix}/bin/perl${perl5.major}" \
                     NO_FINK=1 \
                     NO_DARWIN_PORTS=1 \
                     NO_R_TO_GCC_LINKER=1 \
-                    NO_PERL_CPAN_FALLBACK=1 \
+                    NO_PERL=1 \
                     V=1
 
+if {[variant_isset ${perl5.variant}]} {
+  depends_run-append  port:p${perl5.major}-authen-sasl \
+                      port:p${perl5.major}-error \
+                      port:p${perl5.major}-net-smtp-ssl \
+                      port:p${perl5.major}-term-readkey \
+                      port:p${perl5.major}-cgi
+
+  build.args-delete   NO_PERL=1
+
+  build.args-append   PERL_PATH="${prefix}/bin/perl${perl5.major}" \
+                      NO_PERL_CPAN_FALLBACK=1
+} else {
+  if {[variant_isset svn]} {
+    error "+svn does not work without +perl5_xx"
+  }
+}
```

Comment:
The indentation of your changes does not match the rest of the portfile or conform to the modeline. Indentations shall be at multiples of 4 spaces.

Ports should use `return -code error` rather than `error`. And you cannot issue an error outside of a phase, else it would also occur for users just running something innocuous like `port info` or `port clean`.

You don't need to write `${prefix}/bin/perl${perl5.major}`; the perl5 portgroup defines the variable `${perl5.bin}` with this value for you.

No need for a second nested `if` statement; it can be an `elseif` clause of the first `if` statement.

```suggestion
if {[variant_isset ${perl5.variant}]} {
    depends_run-append \
                    port:p${perl5.major}-authen-sasl \
                    port:p${perl5.major}-error \
                    port:p${perl5.major}-net-smtp-ssl \
                    port:p${perl5.major}-term-readkey \
                    port:p${perl5.major}-cgi

    build.args-delete \
                    NO_PERL=1

    build.args-append \
                    PERL_PATH="${perl5.bin}" \
                    NO_PERL_CPAN_FALLBACK=1
} elseif {[variant_isset svn]} {
    pre-configure {
        return -code error "+svn does not work without +perl5_xx"
    }
}
```

When selecting a perl variant, the port seems to work fine with these changes.

When installing without a perl variant selected, the port installs some files that have nonsensical shebang lines, for example:

```
$ diff-highlight
-bash: /opt/local/bin/diff-highlight: /opt/local/perl: bad interpreter: No such file or directory
```

This appears to simply be a bug in patch-contrib-diff-highlight-Makefile.diff which does not look like it can ever have been correct. (@Schamschula?) diff-highlight gets the correct MacPorts perl shebang if a perl variant is selected; I don't understand how that works.

If diff-highlight requires MacPorts perl modules, then this variant will also require one of the perl variants like the svn variant does. Or if diff-highlight works with macOS perl without needing additional modules (I think this works), then the patch could be removed.

The /opt/local/share/git/contrib/diff-highlight/ folder also gets installed; it's not clear why, except that the README is useful. But installing the Makefile and tests and so forth for diff-highlight does not seem useful when we already install the program.

Reply:
I have just checked the code for `diff-highlight`, and it definitely requires Perl to run (it is a Perl script). Although it probably runs using only macOS Perl, doesn't that go slightly against the self contained philosophy of MacPorts using its own toolchain? IMO, if one explicitly disables Perl support, one should probably expect `diff-highlight` to stop working and this option should error out the same way as `svn`.

That being said, I think I could probably make the patch work in such a way that it uses MacPorts Perl if installed with Perl support, otherwise macOS Perl if not, so please let me know if you'd rather have it completely disabled (like `svn`) or have it fallback on macOS Perl. Thank you!

Reply:
It's acceptable to use macOS perl if you don't require any additional modules.

I think patch-contrib-diff-highlight-Makefile.diff is not needed and removing it would accomplish the goal.

Reply:
I have double checked the Makefile manual, and you are right, [variables specified on make command line override the values assigned in makefile](https://www.gnu.org/software/make/manual/make.html#Overriding), so will remove the patch and add your suggestions, thanks!

Reply:
Forgot to write that I have tested `diff-highlight` and it works with macOS Perl!